types/benches/checksummed_hex_bench.rs

```diff
@@ -0,0 +1,104 @@
+use casper_types::checksummed_hex::{decode, encode, encode_iter};
+use criterion::{black_box, criterion_group, criterion_main, Criterion};
+
+fn encode_and_print_first_5(input: &[u8]) {
+    let encoded = encode(input);
+    format!("{:5}", encoded);
+}
+
+fn encode_iter_and_print_first_5(input: &[u8]) {
+    let encoded = encode_iter(input).take(5).collect::<String>();
+    format!("{:5}", encoded);
+}
+
+fn encode_iter_collected(input: &[u8]) -> String {
+    encode_iter(input).collect()
+}
+
+fn encode_vs_encode_iter(c: &mut Criterion) {
+    let input: Vec<u8> = (0..100_000).map(|_| rand::random::<u8>()).collect();
+    let mut group = c.benchmark_group("encode_vs_encode_iter");
+
+    group.bench_function("encode", |b| b.iter(|| encode(black_box(&input))));
+    group.bench_function("encode_iter_collected", |b| {
+        b.iter(|| encode_iter_collected(black_box(&input)))
+    });
+    group.finish();
+}
+
+fn encode_vs_encode_iter_first_5_chars(c: &mut Criterion) {
+    let input: Vec<u8> = (0..100_000).map(|_| rand::random::<u8>()).collect();
+    let mut group = c.benchmark_group("encode_vs_encode_iter_first_5");
+    group.bench_function("encode_first_5", |b| {
+        b.iter(|| encode_and_print_first_5(black_box(&input)))
+    });
+    group.bench_function("encode_iter_first_5", |b| {
+        b.iter(|| encode_iter_and_print_first_5(black_box(&input)))
+    });
+    group.finish();
+}
+
+fn encode_vs_base16_encode_lower_10kb(c: &mut Criterion) {
+    let input: Vec<u8> = (0..10_000).map(|_| rand::random::<u8>()).collect();
+    let mut group = c.benchmark_group("encode_vs_encode_lower");
+    group.bench_function("encode", |b| b.iter(|| encode(black_box(&input))));
+    group.bench_function("base16_encode_lower", |b| {
+        b.iter(|| base16::encode_lower(black_box(&input)))
+    });
+    group.finish();
+}
+
+fn encode_vs_base16_encode_lower_100kb(c: &mut Criterion) {
+    let input: Vec<u8> = (0..100_000).map(|_| rand::random::<u8>()).collect();
+    let mut group = c.benchmark_group("encode_vs_encode_lower");
```

Comment:
Do we need to update the name of the group to reflect that it's done on `100kb` input? Would that information be available in the resulting report otherwise?

Reply:
Updated.